### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by nashit-mashkoor**
The following branch contains two features:
1)User authentication
2)Profile picture addition and deletion

### CODE REVIEW

File: Internship/.generators [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (shaffan-chaudhry-s63) [inline Internship/.generators:1]: What is this file?

File: Internship/.rakeTasks [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (shaffan-chaudhry-s63) [inline Internship/.rakeTasks:1]: What is this file?

File: Internship/IMDb.iml [hunk 3]
```diff
@@ -0,0 +1,127 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (shaffan-chaudhry-s63) [inline Internship/IMDb.iml:1]: What is this file?

File: Internship/dictionaries/square63.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<component name="ProjectDictionaryState">
```
Comment (shaffan-chaudhry-s63) [inline Internship/dictionaries/square63.xml:1]: What is this file?

File: Internship/misc.xml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (shaffan-chaudhry-s63) [inline Internship/misc.xml:1]: What is this file?

File: app/assets/javascripts/application.js [hunk 6]
```diff
@@ -12,5 +12,10 @@
 //
 //= require rails-ujs
 //= require activestorage
-//= require turbolinks
+
```
Comment (shaffan-chaudhry-s63) [inline app/assets/javascripts/application.js:15]: Remove spaces in between require

File: app/assets/javascripts/application.js [hunk 7]
```diff
@@ -12,5 +12,10 @@
...
+//= require jquery
+//= require bootstrap-sprockets
+
+//= require jquery.turbolinks
```
Comment (shaffan-chaudhry-s63) [inline app/assets/javascripts/application.js:19]: We are not using turbolinks so remove this.

File: app/assets/stylesheets/application.css.scss [hunk 8]
```diff
@@ -13,3 +13,17 @@
  *= require_tree .
  *= require_self
  */
+@import "bootstrap-sprockets";
+@import "bootstrap";
+@import 'bootstrap/theme';
+
+
```
Comment (shaffan-chaudhry-s63) [inline app/assets/stylesheets/application.css.scss:20]: There should be one line in between.

File: app/helpers/application_helper.rb [hunk 9]
```diff
@@ -1,2 +1,15 @@
 module ApplicationHelper
+  # Returns the Gravatar for the given user.
+
+  def user_profile_pic(user, size= 40)
+
```
Comment (shaffan-chaudhry-s63) [inline app/helpers/application_helper.rb:5]: remove this extra line.

File: app/helpers/application_helper.rb [hunk 10]
```diff
@@ -1,2 +1,15 @@
...
+    if user.profile_pic.attached?
+      user.profile_pic.variant(resize: "#{size}x#{size}!")
+    else
+      gravatar_id = 1
+      return "https://secure.gravatar.com/avatar/#{gravatar_id}?s=#{size}"
+    end
+
```
Comment (shaffan-chaudhry-s63) [inline app/helpers/application_helper.rb:12]: remove this extra line

File: app/helpers/application_helper.rb [hunk 11]
```diff
@@ -1,2 +1,15 @@
...
+  end
+
```
Comment (shaffan-chaudhry-s63) [inline app/helpers/application_helper.rb:14]: Remove this extra line

File: app/models/user.rb [hunk 12]
```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  validates :name, presence: true
+  has_one_attached :profile_pic
+
+  attr_accessor :remove_profile_pic
```
Comment (shaffan-chaudhry-s63) [inline app/models/user.rb:9]: Move this to the top

File: app/models/user.rb [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+
+  after_save :purge_profile_pic, if: :remove_profile_pic
+  private def purge_profile_pic
```
Comment (shaffan-chaudhry-s63) [inline app/models/user.rb:12]: correct this styling

File: app/views/devise/confirmations/new.html.erb [hunk 14]
```diff
@@ -0,0 +1,16 @@
+<h2>Resend confirmation instructions</h2>
+
+<%= form_for(resource, as: resource_name, url: confirmation_path(resource_name), html: { method: :post }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email", value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email) %>
```
Comment (shaffan-chaudhry-s63) [inline app/views/devise/confirmations/new.html.erb:8]: Use single quoted strings in rails code.

File: app/views/devise/confirmations/new.html.erb [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Resend confirmation instructions" %>
```
Comment (shaffan-chaudhry-s63) [inline app/views/devise/confirmations/new.html.erb:12]: Use single quoted strings in rails code.

File: app/views/devise/passwords/edit.html.erb [hunk 16]
```diff
@@ -0,0 +1,25 @@
+<h2>Change your password</h2>
+
+<%= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+  <%= f.hidden_field :reset_password_token %>
+
+  <div class="field">
+    <%= f.label :password, "New password" %><br />
```
Comment (shaffan-chaudhry-s63) [inline app/views/devise/passwords/edit.html.erb:8]: Use single quoted strings in rails code.

File: app/views/devise/passwords/edit.html.erb [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+    <% if @minimum_password_length %>
+      <em>(<%= @minimum_password_length %> characters minimum)</em><br />
+    <% end %>
+    <%= f.password_field :password, autofocus: true, autocomplete: "new-password" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password_confirmation, "Confirm new password" %><br />
```
Comment (shaffan-chaudhry-s63) [inline app/views/devise/passwords/edit.html.erb:16]: Use single quoted strings in rails code.

File: app/views/devise/passwords/edit.html.erb [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+    <%= f.password_field :password_confirmation, autocomplete: "off" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Change my password" %>
```
Comment (shaffan-chaudhry-s63) [inline app/views/devise/passwords/edit.html.erb:21]: Use single quoted strings in rails code.

File: app/views/devise/passwords/edit.html.erb [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (shaffan-chaudhry-s63) [inline app/views/devise/passwords/edit.html.erb:25]: Use single quoted strings in rails code.

File: app/views/devise/registrations/edit.html.erb [hunk 20]
```diff
@@ -0,0 +1,60 @@
...
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :name%><br />
+    <%= f.text_field :name, autofocus: true, autocomplete: "name"  %>
+  </div>
+
+  <div class="field">
+    <%= f.label :profile_pic%><br />
+    <%= image_tag user_profile_pic(resource,100), class: 'img-circle' %><br />
```
Comment (shaffan-chaudhry-s63) [inline app/views/devise/registrations/edit.html.erb:13]: Use space after comma: `, 100)`

File: app/views/layouts/_navbar.html.erb [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+    </div>
+    <div id="navbar">
+      <ul class="nav navbar-nav">
+        <li><%= link_to 'Home', root_path %></li>
+      </ul>
+      <ul class="nav navbar-nav pull-right">
+        <% if user_signed_in? %>
+          <li class="dropdown">
+            <a class="dropdown-toggle" data-toggle="dropdown" href="#">
+              <%= image_tag user_profile_pic(current_user,35), class: 'img-circle' %>
```
Comment (shaffan-chaudhry-s63) [inline app/views/layouts/_navbar.html.erb:14]: Use space after comma: `, 35)`

File: test/controllers/pages_controller_test.rb [hunk 22]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
```
Comment (shaffan-chaudhry-s63) [inline test/controllers/pages_controller_test.rb:1]: Remove this file

File: test/fixtures/users.yml [hunk 23]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (shaffan-chaudhry-s63) [inline test/fixtures/users.yml:1]: Remove this file

File: test/models/user_test.rb [hunk 24]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (shaffan-chaudhry-s63) [inline test/models/user_test.rb:1]: Remove this file.

**Review by shaffan-chaudhry-s63 — CHANGES_REQUESTED** [review]
Incorporate change as requested in comments.

File: Internship/.generators (hunk 1 above)
Comment (nashit-mashkoor) [reply Internship/.generators:1]: Removed

File: Internship/.rakeTasks (hunk 2 above)
Comment (nashit-mashkoor) [reply Internship/.rakeTasks:1]: Removed

File: Internship/IMDb.iml (hunk 3 above)
Comment (nashit-mashkoor) [reply Internship/IMDb.iml:1]: Removed

File: Internship/dictionaries/square63.xml (hunk 4 above)
Comment (nashit-mashkoor) [reply Internship/dictionaries/square63.xml:1]: Removed

File: Internship/misc.xml (hunk 5 above)
Comment (nashit-mashkoor) [reply Internship/misc.xml:1]: removed

File: app/views/devise/confirmations/new.html.erb (hunk 14 above)
Comment (nashit-mashkoor) [reply app/views/devise/confirmations/new.html.erb:8]: This has been generated by the devise generator

File: app/views/devise/confirmations/new.html.erb (hunk 15 above)
Comment (nashit-mashkoor) [reply app/views/devise/confirmations/new.html.erb:12]: Generated by Devise

File: app/views/devise/passwords/edit.html.erb (hunk 16 above)
Comment (nashit-mashkoor) [reply app/views/devise/passwords/edit.html.erb:8]: Generated by Devise

**Pull request closed without merging**